airflow/jobs/scheduler_job.py

```diff
@@ -395,6 +401,15 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
                         dag_id,
                         max_active_tasks_per_dag_limit,
                     )
+                    # update the last_scheduling_decision for all task instances in this dag
+                    session.query(TI).filter(
+                        TI.dag_id == task_instance.dag_id, TI.state == State.SCHEDULED
+                    ).update(
+                        {
+                            TI.last_scheduling_decision: timezone.utcnow(),
+                        },
+                        synchronize_session=False,
```

Comment:
This update is overly broad -- it is going to update _every single task for this DAG_, even completed ones.

Reply:
My comment here was wrong, it actually updates only those in scheduled state